sasmodels/TwoYukawa/CalcRealRoot.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+
+from .Epoly import make_poly
+
+def CalRealRoot(coeff):
+
+    Ecoefficient = coeff.Ecoefficient
+
+    # Ick! Code is duplicated in Epoly.py
+    gE1d20 = Ecoefficient[0]
+    gE1d11 = Ecoefficient[1]
+    gE1d21 = Ecoefficient[2]
+    gE1d31 = Ecoefficient[3]
+    gE1d02 = Ecoefficient[4]
+    gE1d12 = Ecoefficient[5]
+    gE1d22 = Ecoefficient[22]
+    gE1d32 = Ecoefficient[23]
+    gE1d42 = Ecoefficient[6]
+    gE1d13 = Ecoefficient[7]
+    gE1d23 = Ecoefficient[8]
+    gE1d33 = Ecoefficient[9]
+    gE1d24 = Ecoefficient[10]
+
+    gE2d20 = Ecoefficient[11]
+    gE2d11 = Ecoefficient[12]
+    gE2d21 = Ecoefficient[13]
+    gE2d31 = Ecoefficient[14]
+    gE2d02 = Ecoefficient[15]
+    gE2d12 = Ecoefficient[16]
+    gE2d22 = Ecoefficient[24]
+    gE2d32 = Ecoefficient[25]
+    gE2d42 = Ecoefficient[17]
+    gE2d13 = Ecoefficient[18]
+    gE2d23 = Ecoefficient[19]
+    gE2d33 = Ecoefficient[20]
+    gE2d24 = Ecoefficient[21]
+
+    d2Coeff = make_poly(Ecoefficient)
+    # print("Polynomial coefficients:"); print(d2Coeff)
+    # print("Poly: "+ " ".join(f"{coeff:+8.1e}" for coeff in d2Coeff))
+
+    N=len(d2Coeff)
+
+    if 1:
+        Factor = 1  # 20
+        FactorArray = Factor ** np.arange(N)[::-1]
+        d2Coeff_factor = d2Coeff*FactorArray
+        myd2Root_factor = np.roots(d2Coeff_factor)
+        myd2Root = myd2Root_factor*Factor
+    else:
+        # Very slow high precision root finder
+        import mpmath
+        myd2Root = mpmath.polyroots(d2Coeff, maxsteps=500, extraprec=1000)
+
+    if 0:
+        import mpmath
+        myd2Root_mpmath = mpmath.polyroots(d2Coeff, maxsteps=500, extraprec=1000)
+        nproots = list(sorted(myd2Root, key=lambda v: (v.real, v.imag)))
+        # print(nproots/Factor)
+        mproots = list(complex(v) for v in sorted(myd2Root_mpmath, key=lambda v: (v.real,v.imag)))
+        for k, (npr, mpr) in enumerate(zip(nproots, mproots)):
+            if (mpr.imag == 0 and mpr != 0):
+                print(f"{k:02} mp:{float(mpr.real):.2f} np:{npr.real:.2f} {abs((npr-mpr)/(abs(mpr) + (mpr==0))):8.2e}")
```

Comment:
All unreachable code. Is this tied to the customize mp methods described in the model file?